hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -967,6 +968,7 @@ private synchronized void removeVolumes(
     // Remove volumes and block infos from FsDataset.
     data.removeVolumes(storageLocations, clearFailure);
 
+    Preconditions.checkNotNull(data, "Storage not yet initialized");
```

Comment:
Maybe this check should be before `data.removeVolumes(storageLocations, clearFailure); `?

Reply:
> Maybe this check should be before `data.removeVolumes(storageLocations, clearFailure); `?

I'm sorry. It was my fault and I fixed it.